test/integ/test_multi_statement.py

```diff
@@ -0,0 +1,268 @@
+            assert con.is_an_error(status)
+            with pytest.raises(ProgrammingError) as e1:
+                con.get_query_status_throw_if_error(q_id)
+            assert sync_error.value.errno != -1
+            with pytest.raises(ProgrammingError) as e2:
+                cur.get_results_from_sfqid(q_id)
+            assert e1.value.errno == e2.value.errno == sync_error.value.errno
+
+
+@pytest.mark.skipolddriver
+def test_mix_sync_async_multi(conn_cnx):
+    """Tests sending multiple multi-statement async queries at the same time."""
+    with conn_cnx() as con:
+        with con.cursor() as cur:
+            # Setup
+            cur.execute("alter session set MULTI_STATEMENT_COUNT = 0")
+            cur.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_TZ")
+            try:
+                cur.execute(
+                    "create or replace table smallTable (colA string, colB int);"
+                    "create or replace table uselessTable (colA string, colB int);"
+                )
+                for table in ["smallTable", "uselessTable"]:
+                    cur.execute(
+                        "insert into {t} values('row1', 1);"
+                        "insert into {t} values('row2', 2);"
+                        "insert into {t} values('row3', 3);".format(t=table)
+                    )
+                cur.execute_async("select 1; select 'a'; select * from smallTable;")
+                sf_qid1 = cur.sfqid
+                cur.execute_async("select 2; select 'b'; select * from uselessTable")
+                sf_qid2 = cur.sfqid
+                # Wait until the 2 queries finish
+                while con.is_still_running(con.get_query_status(sf_qid1)):
+                    time.sleep(1)
+                while con.is_still_running(con.get_query_status(sf_qid2)):
+                    time.sleep(1)
+                cur.execute("drop table uselessTable")
+                assert cur.fetchall() == [("USELESSTABLE successfully dropped.",)]
+                cur.get_results_from_sfqid(sf_qid1)
+                assert cur.fetchall() == [(1,)]
+                assert cur.nextset() == cur
+                assert cur.fetchall() == [("a",)]
+                assert cur.nextset() == cur
+                assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+                assert cur.nextset() is None
+                assert not cur.fetchall()
+                cur.get_results_from_sfqid(sf_qid2)
+                assert cur.fetchall() == [(2,)]
+                assert cur.nextset() == cur
+                assert cur.fetchall() == [("b",)]
+                assert cur.nextset() == cur
+                assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+                assert cur.nextset() is None
+                assert not cur.fetchall()
+            finally:
+                for table in ["smallTable", "uselessTable"]:
+                    cur.execute(f"drop table if exists {table}")
```

Comment:
You can get rid of this by making these tables temporary https://docs.snowflake.com/en/user-guide/tables-temp-transient.html#temporary-tables